packages/studioCMS/src/integrations/studioCMSDashboard/routes/dashboard/pages/edit/page-list.astro

```diff
@@ -0,0 +1,121 @@
+---
+// @ts-expect-error - Some types can only be imported from the Astro runtime
+import { SiteConfig, db, eq, asc, PageData } from 'astro:db';
+import { Layout, FormattedDate } from 'studiocms-dashboard:components';
+import Config from 'virtual:studiocms/config';
+import { authHelper, pluginList, type Locals } from 'studiocms:helpers';
+import { StudioCMSRoutes, getEditRoute, getDeleteRoute } from 'studiocms-dashboard:routeMap';
+import type { SiteConfigResponse, pageDataReponse } from 'studiocms:components';
+
+const { dashboardConfig: { developerConfig: { testingAndDemoMode } } } = Config;
+
+const { 
+    authLinks: { 
+        loginURL 
+    }, 
+    mainLinks: { 
+        userProfile,
+        dashboardIndex,
+        pageEdit,
+    } 
+} = StudioCMSRoutes;
+
+const contextConfig: SiteConfigResponse = await db.select().from(SiteConfig).where(eq(SiteConfig.id, 1)).get();
+const pages: pageDataReponse[] = await db.select().from(PageData).orderBy(asc(PageData.id));
+const { permissionLevel } = await authHelper(Astro.locals as Locals);
+
+// If the user is not logged in, redirect to the login page
+if ( !testingAndDemoMode ) {
+    if ( permissionLevel === "unknown" ) {
+        console.log('User is not logged in. Redirecting to login page.');
```

Comment:
It's worth wrapping these console logs in your console verbose handler dealy

Reply:
I cant use the `AstroIntegrationLogger` outside of the integration config itself.... right? so that would require a runtime version?

Reply:
Hmmm I thought you had built a custom logger? Or does that use `AstroIntegrationLogger`?

Reply:
it uses the `AstroIntegrationLogger`, i did build one for Astro-gists that uses console.log with `Kleur`, similar to how astro internals work... i could build another "runtime" version of that for `console.logs` from the actual dashboard since it is SSR

Reply:
Hmmm, leave it for now but maybe one to come back to!